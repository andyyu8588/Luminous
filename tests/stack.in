import Stack

class Object {
    public constructor() {
        print("Object constructed");
    }
}

stack = Stack();
stack.push(Object());
stack.push("Hello world!");
stack.push(3.1415926);
print(stack.isEmpty());
print(stack.size());
print(stack.pop());
print(stack.size());
print(stack.top());
stack.pop();
print(stack.size());
print(stack.pop());
print(stack.isEmpty());
print(stack.size());