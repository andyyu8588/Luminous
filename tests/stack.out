Object constructed
false
3
3.14159
2
Hello world!
1
Object instance
true
0
